overlord/overlord_test.go

```diff
@@ -112,3 +116,55 @@ func (os *overlordSuite) TestEnsureLoopRunAndStop(c *C) {
 
 	c.Check(ensureCalls >= 2, Equals, true)
 }
+
+func (ovs *overlordSuite) TestEnsureLoopMediatedEnsureAfter(c *C) {
```

Comment:
Can we have a test making sure EnsureAfter(0) works as expected when inside an Ensure call, rescheduling immediately rather than blowing up or scheduling on the default interval?